AnomalyDetection.py

```diff
@@ -26,98 +29,88 @@ def _ComputeScore(coefs, values):
         residual += deviation * deviation
     return math.sqrt(residual / len(values)), slope
 
-def DetectAnomalies(data, windowSize, levels=1, numTopResults=None, visualize=False):
 
+#TODO ktran: improve the algorithm
+def _partition_anomalies(windows, k):
+    """
+    :param windows: windows, sorted by anomaly score in descending order
+    :param k: number of partitions
+    :return: partition positions
+    """
+    diffs = [windows[iw - 1][1] - windows[iw][1]
+             for iw in range(1, len(windows))]
+    top_jump_positions = sorted(range(len(diffs)), key=lambda i: diffs[i], reverse=True)[0:k]
+    return sorted(top_jump_positions, reverse=True)
+
+
+def detect_anomalies(data, window_size, sensitivity=5, num_anomalies=None, visualize=True):
     if type(data) != pd.Series:
         raise ValueError('data must be of the pandas Series type')
+    data = data.fillna(0)  #fill NANs with 0 to make the series contiguous
```

Comment:
consider padding instead of 0.

Reply:
Padding by 0 is by design. That will also likely trigger an alert when the XPert suddenly has a data loss problem.

Reply:
On the second thought, I think pad makes more sense. I was looking at an example and I saw too many interruptions in Xpert over some periods. We can expose that as an option but `pad` as default value makes more sense.